src/CoreWCF.WebHttp/src/CoreWCF/UriTemplateTable.cs

```diff
@@ -0,0 +1,616 @@
+            //  settings and the matching results. We will assume that the 'best' matches will be the ones with
+            //  the smallest number of segments - this will prefer ".../" over ".../{x=1}[/...]".
+            if (NotAllCandidatesArePathFullyEquivalent(candidates))
+            {
+                Collection<UriTemplateTableMatchCandidate> nextCandidates = new Collection<UriTemplateTableMatchCandidate>();
+                int minSegmentsCount = -1;
+                for (int i = 0; i < candidates.Count; i++)
+                {
+                    UriTemplateTableMatchCandidate candidate = candidates[i];
+                    if (minSegmentsCount == -1)
+                    {
+                        minSegmentsCount = candidate.Template._segments.Count;
+                        nextCandidates.Add(candidate);
+                    }
+                    else if (candidate.Template._segments.Count < minSegmentsCount)
+                    {
+                        minSegmentsCount = candidate.Template._segments.Count;
+                        nextCandidates.Clear();
+                        nextCandidates.Add(candidate);
+                    }
+                    else if (candidate.Template._segments.Count == minSegmentsCount)
+                    {
+                        nextCandidates.Add(candidate);
+                    }
+                }
+
+                Fx.Assert(minSegmentsCount != -1, "At least the first entry in the list should be kept");
+                Fx.Assert(nextCandidates.Count >= 1, "At least the first entry in the list should be kept");
+                Fx.Assert(nextCandidates[0].Template._segments.Count == minSegmentsCount, "Trivial");
+                candidates = nextCandidates;
+            }
+
+            // Building the actual result
+            Collection<UriTemplateMatch> actualResults = new Collection<UriTemplateMatch>();
+            for (int i = 0; i < candidates.Count; i++)
+            {
+                UriTemplateTableMatchCandidate candidate = candidates[i];
+                UriTemplateMatch match = candidate.Template.CreateUriTemplateMatch(_originalUncanonicalizedBaseAddress,
+                    uri, candidate.Data, candidate.SegmentsCount, relativeSegments, queryParameters);
+                actualResults.Add(match);
+            }
+
+            return actualResults;
+        }
+        public UriTemplateMatch MatchSingle(Uri uri)
+        {
+            Collection<UriTemplateMatch> c = Match(uri);
+            if (c.Count == 0)
+            {
+                return null;
+            }
+            if (c.Count == 1)
+            {
+                return c[0];
+            }
+            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new UriTemplateMatchException(SR.UTTMultipleMatches));
+        }
+
+        static bool AllEquivalent(IList<UriTemplateTableMatchCandidate> list, int a, int b)
+        {
```

Comment:
Missing accessibility modifiers in multiple places in this file.

Reply:
Addressed in 14ac88e67a220d805d2fac57fa31026cd397ba1e.